modules/protocol/resources/META-INF/smithy/smithy4s.smithy

```diff
@@ -41,3 +41,7 @@ structure UncheckedExample {
 @trait(selector: "string")
 structure uuidFormat {
 }
+
+@trait(selector: "union")
+structure discriminated {
```

Comment:
I think this should be parameterized by the field name. Personally I've seen "@type", "type", "kind", "node_kind" and so on, so it should probably be up to the API drafter :)

Reply:
That is indeed correct. See "propertyName" in openapi : https://spec.openapis.org/oas/latest.html#discriminator-object. 

I think I've seen `type` more than anything else, but making it explicit is probably a good call.